server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -481,4 +482,51 @@ public void onFailure(Exception e) {
             }
         };
     }
+
+    public void deleteDecommissionAttribute(final ActionListener<AcknowledgedResponse> listener) {
+        decommissionController.clearVotingConfigExclusion(new ActionListener<Void>() {
+            @Override
+            public void onResponse(Void unused) {
+                logger.info("successfully cleared voting config exclusion for delting the decommission");
+                clusterUpdateTaskForDeletingDecommission(listener);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.debug(new ParameterizedMessage("failure in clearing voting config during delete_decommission request"), e);
+                listener.onFailure(e);
+            }
+        }, false);
+    }
+
+    void clusterUpdateTaskForDeletingDecommission(ActionListener<AcknowledgedResponse> listener) {
+        clusterService.submitStateUpdateTask("delete_decommission_state", new ClusterStateUpdateTask(Priority.URGENT) {
+            @Override
+            public ClusterState execute(ClusterState currentState) {
+                return clearDecommissionedAttributeFromMetadata();
+            }
+
+            @Override
+            public void onFailure(String source, Exception e) {
+                logger.error(() -> new ParameterizedMessage("Failed to clear decommission attribute."), e);
+                listener.onFailure(e);
+            }
+
+            @Override
+            public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
+                // Cluster state Processed for deleting the decommission attribute.
+                assert newState.metadata().decommissionAttributeMetadata() == null;
+                listener.onResponse(new AcknowledgedResponse(true));
+            }
+        });
+    }
+
+    ClusterState clearDecommissionedAttributeFromMetadata() {
+        logger.info("Deleting the decommission attribute from cluster state");
+        ClusterState state = clusterService.getClusterApplierService().state();
```

Comment:
Rather than this, accept state from execute method here. That state is source of truth during state update tasks

Reply:
Done.